web/lib/api/metrics/averageTimeToFirstToken.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  FilterNode,
+  timeFilterToFilterNode,
+} from "../../../services/lib/filters/filterDefs";
+import { buildFilterWithAuthClickHouse } from "../../../services/lib/filters/filters";
+import { Result, resultMap } from "../../result";
+import { dbQueryClickhouse } from "../db/dbExecute";
+
+export async function getAverageTimeToFirstToken(
+  filter: FilterNode,
+  timeFilter: {
+    start: Date;
+    end: Date;
+  },
+  org_id: string
+): Promise<Result<number, string>> {
+  const { filter: filterString, argsAcc } = await buildFilterWithAuthClickHouse(
+    {
+      org_id,
+      filter: {
+        left: timeFilterToFilterNode(timeFilter, "request_response_log"),
+        right: filter,
+        operator: "and",
+      },
+      argsAcc: [],
+    }
+  );
+  const query = `
+  WITH total_count AS (
+    SELECT 
+      count(*) as count,
+      sum(request_response_log.time_to_first_token) as total_time_to_first_token
+    FROM request_response_log
+    WHERE (
+      (${filterString})
+    )
+  )
+  SELECT CASE
+    WHEN count = 0 THEN 0
+    ELSE total_time_to_first_token / count
+  END as average_time_to_first_token
+  FROM total_count
```

Comment:
Can we just select the average here? 

This query is resulting in some weird calcs:


<img width="553" alt="image" src="https://github.com/Helicone/helicone/assets/26822232/75336177-52f2-4bc5-abb9-2b2b63dd1192">

Reply:
SELECT avg (request_response_log.time_to_first_token)

Where request_response_log.time_to_first_token > 0

Reply:
something like that ^

Reply:
@chitalian , It's query for metric value (not overtime). I've fixed overtime and removed distinct